src/lib/merkle_ledger/location.ml

```diff
@@ -147,6 +147,20 @@ module T = struct
     | Right ->
         (sibling, base)
 
+  let merkle_path_dependencies_exn (location : t) : (t * Direction.t) list =
+    let rec loop k acc =
+      if Addr.depth k = 0 then acc
+      else
+        let sibling = Hash (Addr.sibling k) in
+        let sibling_dir = last_direction k in
```

Comment:
This should be `let sibling_dir = last_direction (Addr.sibling k) in` because we are trying to find the direction of sibling instead of the address itself.